src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -211,7 +211,7 @@ private static async Task<string> ReadAsStringAsync(Stream s)
             // This leads to the inability to use the stream further by other consumers/middlewares of the request processing
             // pipeline. In fact, it is absolutely not dangerous not to dispose StreamReader as it does not perform any useful
             // work except for the disposing inner stream.
-            return await new StreamReader(s).ReadToEndAsync();
+            return await new StreamReader(s).ReadToEndAsync().ConfigureAwait(false); ;
```

Comment:
this can be a direct return to avoid the async state machine

```
        private static Task<string> ReadAsStringAsync(Stream s)
        {
            // Do not explicitly or implicitly (via using, etc.) call dispose because StreamReader will dispose inner stream.
            // This leads to the inability to use the stream further by other consumers/middlewares of the request processing
            // pipeline. In fact, it is absolutely not dangerous not to dispose StreamReader as it does not perform any useful
            // work except for the disposing inner stream.
            return new StreamReader(s).ReadToEndAsync();
        }
```

Reply:
i will do this in a new PR after this is merged